### PR TITLE
Small swig changes

### LINKED ISSUE #104: Support pathlib filespecs
As Python's `pathlib` gains popularity, it would be nice if `cspyce` accepted `pathlib` objects where filenames are expected.

See `SpiceyPy`: https://github.com/AndrewAnnex/SpiceyPy/pull/436

### PR DESCRIPTION
**Pull request by fyellin**
FIXES #104 

1) Add more typecasts to eliminate pointer warnings.
2) Add more filepaths and test them.
3) Clean up some code that should have been cleaned up earlier: SWIG knows how to clean up mallocs when a failure occurs.

### CODE REVIEW

File: unittests/test_a_c.py [hunk 1]
```diff
@@ -8,10 +8,10 @@
     CoreKernels,
     CassiniKernels,
     ExtraKernels,
-    download_kernels,
+    checking_pathlike_filename_variants, download_kernels,
```
Comment (rfrenchseti) [inline unittests/test_a_c.py:11]: Please put these on separate lines in all files

File: unittests/test_e.py [hunk 2]
```diff
@@ -721,10 +725,11 @@ def test_ekops():
...
 
 
-def test_ekopw():
-    ekpath = os.path.join(TEST_FILE_DIR, "example_ekopw.ek")
+@checking_pathlike_filename_variants("path_type_variant")
+def test_ekopw(path_type_variant):
+    ekpath = path_type_variant(os.path.join(TEST_FILE_DIR, "example_ekopw.ek"))
     cleanup_kernel(ekpath)
-    handle = cs.ekopn(ekpath, ekpath, 80)
+    handle = cs.ekopn(ekpath, str(ekpath), 80)
```
Comment (rfrenchseti) [inline unittests/test_e.py:732]: All of the other tests create a string and then make a path variant out of it, then use the string later. This one converts the path back to a string. Is that intentional?

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
Could you update the README?

First, the following is out of date and should just be removed:

```
The CSPICE concepts of "windows" and "cells" are not needed in Python. In CSPICE, these allow a function to return a variable amount of information, which the program can then iterate through. The cspyce counterpart of each of these functions simply returns a complete list of the results.
```

Second, add a line saying that functions that take filenames can take variants. You can basically replace the above paragraph with this new one.

File: unittests/test_a_c.py (hunk 1 above)
Comment (fyellin) [reply unittests/test_a_c.py:11]: Done.  My bad for letting PyCharm handle the imports without paying attention.

File: unittests/test_e.py (hunk 2 above)
Comment (fyellin) [reply unittests/test_e.py:732]: You're right.  This one was updated before I had updated the others and ran into a problem.

Given this use of ekopn(), we might want to re-consider our decision that "internal names" are strings.  Not sure how ekopn (or internal names) are used in general.

**Comment by fyellin** [comment]
Readme updated.  Other changes done.

**Review by rfrenchseti — APPROVED** [review]

**Pull request merged**